hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -181,15 +193,17 @@ public BackgroundTaskResult call() throws Exception {
           List<Pair<String, String>> prefixPairs =
               constructPrefixPairs(snapshotInfo);
 
-          try (ReferenceCounted<IOmMetadataReader, SnapshotCache>
-                   snapshotMetadataReader = snapshotCache.get()
-              .get(snapshotInfo.getTableKey())) {
+          try (
+              ReferenceCounted<IOmMetadataReader, SnapshotCache>
+                  snapshotMetadataReader = snapshotCache.get()
+                  .get(snapshotInfo.getTableKey(), true)) {
```

Comment:
I'll suggested to use `if (snapshotInfo.isSstFiltered() || snapshotInfo.getSnapshotStatus() != SnapshotInfo.SnapshotStatus.SNAPSHOT_DELETED)` at line number 186 instead of loading DELETED snapshot.

And add a catch at line 209:
```
          } catch (OMException omException) {
            if (snapshotInfo.getSnapshotStatus() ==
                SnapshotInfo.SnapshotStatus.SNAPSHOT_DELETED) {
              LOG.info("Snapshot with name: `{}`, id: `{}` has been deleted.",
                  snapshotInfo.getName(), snapshotInfo.getSnapshotId());
            }
          }
```

Reply:
Done.